### ISSUE TITLE
Segmentation fault upon expanding basecamp to include mechanic shop

### ISSUE THREAD

**Issue by MrHrulgin** [CleverRaven/Cataclysm-DDA#67848]
### Describe the bug

I expanded a new camp to include a mechanic shop. It crashed.

### Attach save file

N/A

### Steps to reproduce

I expanded a new camp to include a mechanic shop. It crashed.

### Expected behavior

Not crashing?

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3324 (22H2)
- Game Version: 2c8ae07 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Magiclysm [magiclysm],
    Blaze Industries [blazeindustries],
    Mind Over Matter [mindovermatter],
    Translate Complex Dialogue [translate_dialogue],
    Alternative Map Key [alt_map_key],
    No Rail Stations [No_Rail_Stations],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills]
]

### Additional context

2023-08-23 08:09:34,417 [MainThread  ] [INFO ]  CDDA Game Launcher started: 1.6.9
2023-08-23 08:09:36,264 [MainThread  ] [INFO ]  Checking for new launcher version
The program has crashed.
See the log file for a stack trace.
CRASH LOG FILE: ./config/crash.log
VERSION: 2c8ae07
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:

  #0
    (dbghelp: debug_write_backtrace+0xb5@0,000,7FF,7BE,01C,1C5[cataclysm-tiles.exe+0x5ec,1c5]),
  #1
    (dbghelp: log_crash+0x377@0,000,7FF,7BD,FE1,687[cataclysm-tiles.exe+0x5b1,687]),
  #2
    (dbghelp: signal_handler+0x57@0,000,7FF,7BD,FE1,C17[cataclysm-tiles.exe+0x5b1,c17]),
  #3
    (dbghelp: seh_filter_exe+0x178@0,000,7FF,7BF,4EC,D08[cataclysm-tiles.exe+0x1,abc,d08]),
  #4
    (dbghelp: `__scrt_common_main_seh'::`1'::filt$0+0x16@0,000,7FF,7BF,71E,1E2[cataclysm-tiles.exe+0x1,cee,1e2]),
  #5
    (dbghelp: _C_specific_handler+0xa0@0,000,7FF,7BF,4BD,F44[cataclysm-tiles.exe+0x1,a8d,f44]),
  #6
    (dbghelp: _chkstk+0x11f@0,000,7FF,C02,FF2,3DF[ntdll.dll+0xa2,3df]),
  #7
    (dbghelp: RtlRaiseException+0x484@0,000,7FF,C02,FA1,4B4[ntdll.dll+0x51,4b4]),
  #8
    (dbghelp: KiUserExceptionDispatcher+0x2e@0,000,7FF,C02,FF0,EEE[ntdll.dll+0xa0,eee]),
  #9
    (dbghelp: is_equal+0x8@0,000,7FF,7BF,02C,9A8[cataclysm-tiles.exe+0x1,5fc,9a8]),
  #10
    (dbghelp: basecamp::is_hidden+0xab@0,000,7FF,7BE,D0B,9AB[cataclysm-tiles.exe+0x1,2db,9ab]),
  #11
    (dbghelp: mission_data::add+0x9f@0,000,7FF,7BF,011,B4F[cataclysm-tiles.exe+0x1,5e1,b4f]),
  #12
    (dbghelp: mission_data::add_start+0x51@0,000,7FF,7BF,011,F11[cataclysm-tiles.exe+0x1,5e1,f11]),
  #13
    (dbghelp: basecamp::get_available_missions_by_dir+0xd59@0,000,7FF,7BF,39D,2A9[cataclysm-tiles.exe+0x1,96d,2a9]),
  #14
    (dbghelp: basecamp::get_available_missions+0x1,4bf@0,000,7FF,7BF,39B,A4F[cataclysm-tiles.exe+0x1,96b,a4f]),
  #15
    (dbghelp: iexamine::bulletin_board+0x166@0,000,7FF,7BE,848,426[cataclysm-tiles.exe+0xe18,426]),
  #16
    (dbghelp: game::examine+0xa65@0,000,7FF,7BE,07F,525[cataclysm-tiles.exe+0x64f,525]),
  #17
    (dbghelp: game::examine+0x747@0,000,7FF,7BE,080,457[cataclysm-tiles.exe+0x650,457]),
  #18
    (dbghelp: game::do_regular_action+0x1,1ed@0,000,7FF,7BE,625,A8D[cataclysm-tiles.exe+0xbf5,a8d]),
  #19
    (dbghelp: game::handle_action+0xa65@0,000,7FF,7BE,62F,1B5[cataclysm-tiles.exe+0xbff,1b5]),
  #20
    (dbghelp: do_turn+0xeae@0,000,7FF,7BE,024,35E[cataclysm-tiles.exe+0x5f4,35e]),
  #21
    (dbghelp: WinMain+0xc85@0,000,7FF,7BD,EAC,D75[cataclysm-tiles.exe+0x47c,d75]),
  #22
    (dbghelp: __scrt_common_main_seh+0x106@0,000,7FF,7BF,4BA,B7A[cataclysm-tiles.exe+0x1,a8a,b7a]),
  #23
    (dbghelp: BaseThreadInitThunk+0x14@0,000,7FF,C02,D57,614[KERNEL32.DLL+0x17,614]),
  #24
    (dbghelp: RtlUserThreadStart+0x21@0,000,7FF,C02,FA2,6B1[ntdll.dll+0x52,6b1]),

**Comment by PatrikLundell** [comment]
I'm not aware of the existence of any mechanic shop expansion. There are two versions of a Workshop expansion, as well as two version of a Garage expansion, and you might mean any of these four.

I tried to repeat your problem with a game version from yesterday and failed to do so. I started a new field base camp version 2, and set the companion to survey expansions until I had two workshops and two garages (one of each version), with no issues.

However, if I read the traceback correctly, the error seems to happen when you order something to be done, not on completion of a mission (such as a survey). If that's correct you're not trying to construct an expansion but rather trying to order the construction of something in a base camp, but if that's the case you've omitted the crucial information of which base camp that might be.
